### ISSUE TITLE
Incorrect behaviour from CHEBCOEFFS for 2nd-kind coefficients.

### ISSUE THREAD

**Issue by nickhale** [chebfun/chebfun#1589]
I believe the following should all give the same result. However, those versions which call `chebcoeffs` with the `'kind', 2` flag seems to return incorrect results.
```
>> f = jacpoly(5, .5, .5);
```
```
>> b = cheb2jac(f.coeffs, .5, .5)
b =
   0.000000000000000
   0.000000000000000
  -0.000000000000000
  -0.000000000000001
  -0.000000000000000
   0.999999999999999
```
```
>> b = jaccoeffs(f, .5+eps, .5)
b =
   0.000000000000000
   0.000000000000000
  -0.000000000000000
  -0.000000000000000
  -0.000000000000000
   1.000000000000000
```
```
>> b = jaccoeffs(f, .5, .5)
b =
                   0
   0.000000000000000
                   0
  -0.000000000000000
                   0
   0.451171875000000
```
```
>> b = chebcoeffs(f, 'kind', 2)
b =
   1.0e-15 *
                   0
   0.222044604925031
                   0
  -0.111022302462516
```
```
>> b = chebcoeffs(f, 6, 'kind', 2)
b =
                   0
   0.000000000000000
                   0
  -0.000000000000000
                   0
   0.451171875000000
```

**Comment by nickhale** [comment]
My bad - part of the problem is the difference in scaling between chebU and Jacobi (0.5, .5).

See
```
>> chebpoly(5, 2)
ans =
   chebfun column (1 smooth piece)
       interval       length     endpoint values  
[      -1,       1]        6        -6        6 
vscale = 6.
>> jacpoly(5, .5, .5)
ans =
   chebfun column (1 smooth piece)
       interval       length     endpoint values  
[      -1,       1]        6      -2.7      2.7 
vscale = 2.707031e+00.
```

However, the third and fourth examples above are still incorrect. (The third because jaccoeffs is calling chebcoeffs and not correcting the scaling. The fourth is presumably some error in truncation location.)

**Comment by aaustin141** [comment]
Closed in #1590.
